lib/scope.js

```diff
@@ -75,6 +75,34 @@ module.exports = {
     return trimmed;
   },  // Ends scope.toTrimmedJSON()
 
+  setReportHeading: async function (scope, { scenario }) {
+    if ( env_vars.DEBUG ) { console.log( scenario.pickle.name ); }
+    // Names for files and reports
+    // TODO: Allow scenario ids to have digits in their name
+
+    // With tag names automatically added to langauges, this should work to make
+    // each language unique.
+    let tags = '';
+    for ( let tag of scenario.pickle.tags ) {
+      tags += tag.name.replace(/[^A-Za-z0-9]+/g, '_').replace(/(\P{L})+/gu, '_');
+    }
+    tags = tags.replace(/_$/, '');
+    let new_id = `${ scenario.pickle.name }${ tags }`;
```

Comment:
This line used to be `scenario_name` in the PR previously. I think that's what may be different. I think we also may need to write separate functions to make the scenario heading than for making filenames. Would appreciate thoughts if people have them!

Reply:
We have separated those concerns. See issue named in initial PR comment.